File: src/main/java/pt/joaomneto/titancompanion/adventurecreation/impl/fragments/VitalStatisticsFragment.java
package pt.joaomneto.titancompanion.adventurecreation.impl.fragments;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import pt.joaomneto.titancompanion.R;
import android.support.v4.app.Fragment;
import pt.joaomneto.titancompanion.util.AdventureFragmentRunner;
import pt.joaomneto.titancompanion.adventurecreation.AdventureCreation;

public class VitalStatisticsFragment extends Fragment {


    View rootView;

    public VitalStatisticsFragment() {

    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        rootView = inflater.inflate(R.layout.fragment_adventurecreation_vital_statistics, container, false);

        return rootView;
    }

    @Override
    public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        rootView.findViewById(R.id.buttonRollStats).setOnClickListener((View v) -> {
            ((AdventureCreation) this.getActivity()).rollStats(v);
        });

        rootView.findViewById(R.id.buttonSaveAdventure).setOnClickListener((View v) -> {
            ((AdventureCreation) this.getActivity()).saveAdventure();
        });
    }
}

File: src/main/java/pt/joaomneto/titancompanion/adventurecreation/impl/fragments/VitalStatisticsFragment.kt
package pt.joaomneto.titancompanion.adventurecreation.impl.fragments

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import pt.joaomneto.titancompanion.R
import pt.joaomneto.titancompanion.adventurecreation.AdventureCreation

open class VitalStatisticsFragment : Fragment() {

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        super.onCreate(savedInstanceState)
        return inflater.inflate(R.layout.fragment_adventurecreation_vital_statistics, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        view.findViewById<View>(R.id.buttonRollStats).setOnClickListener { v: View ->
            (this.activity as AdventureCreation).rollStats(
                v
            )
        }

        view.findViewById<View>(R.id.buttonSaveAdventure).setOnClickListener { v: View -> (this.activity as AdventureCreation).saveAdventure() }
    }
}
